File: python/forma1.py
class Heroi:
    ATAQUES = {
        'mago': 'usou magia',
        'guerreiro': 'usou espada',
        'monge': 'usou artes marciais',
        'ninja': 'usou shuriken'
    }

    def __init__(self, nome, idade, tipo):
        self.nome = nome
        self.idade = idade
        self.tipo = tipo

    def atacar(self):
        ataque = self.ATAQUES.get(self.tipo, 'usou um ataque indefinido')
        print(f"O {self.tipo} atacou usando {ataque}")

# Exemplo de uso:
heroi = Heroi("Herói 1", 30, "mago")
heroi.atacar()

File: js/forma1.js
class heroiDaAventura{
    constructor(nome, idade, tipo, ataque){
        this.nome = nome
        this.idade = idade
        this.tipo = tipo
        this.ataque = ataque
    } 
    historia(){
        console.log(`O grande ${this.tipo}, ${this.nome} atacou usando ${this.ataque}`)
    }

}

let acaoDoHeroi = new heroiDaAventura("Vinicius",26,"mago", "magia")

acaoDoHeroi.historia()
